design/extension.go

```diff
@@ -0,0 +1,963 @@
+			Example("s3")
+			Example("git")
+			Example("mlflow-tracker")
+		})
+	tag++
+	Field(tag, "category", String,
+		`Universal service category. Used to classify services into well-known categories of AI/ML services
+(e.g. model store, feature store, distributed training, serving)`, func() {
+			Default("")
+			MaxLength(100)
+			Example("model-store")
+			Example("serving-platform")
+		})
+	tag++
+	Field(tag, "auth_required", Boolean,
+		`Marks a service for which authentication is required. If set, a set of credentials is required
+to access the service; if none of the provided credentials match the scope of the consumer,
+this service will be excluded from queries`, func() {
+			Default(false)
+		})
+	tag++
+	Field(tag, "description", String, "Service description", func() {
+		MaxLength(1000)
+		Default("")
+	})
+	tag++
+	Field(tag, "configuration", MapOf(String, String),
+		`Configuration entries (e.g. configuration values required to configure all clients that connect to
+this service), expressed as set of key-value entries`, func() {
+			Key(func() {
+				Pattern(identifierPattern)
+			})
+			Example(map[string]string{
+				"authentication": "enabled",
+				"api_version":    "v2",
+			})
+		})
+	tag++
+	Field(tag, "status", ExtensionServiceStatus, "Service status")
+	tag++
+	tag++
+	Field(tag, "endpoints", ArrayOf(ExtensionEndpoint), "List of endpoints through which this service can be accessed")
+	tag++
+	Field(tag, "credentials", ArrayOf(ExtensionCredentials), "List of credentials required to access this service")
+	tag++
+})
+
+// Extension service status descriptor
+var ExtensionServiceStatus = Type("ExtensionServiceStatus", func() {
+	tag := 1
+	Field(tag, "registered", String, "The time when the service was registered", func() {
+		Format(FormatDateTime)
+		Default(time.Now().Format(time.RFC3339))
+		Example(time.Now().Format(time.RFC3339))
+	})
+	tag++
+	Field(tag, "updated", String, "The time when the service was last updated", func() {
+		Format(FormatDateTime)
+		Default(time.Now().Format(time.RFC3339))
+		Example(time.Now().Format(time.RFC3339))
+	})
+	tag++
+})
+
+// Extension endpoint descriptor
+var ExtensionEndpoint = Type("ExtensionEndpoint", func() {
+	tag := 1
+	Field(tag, "url", String,
+		`Endpoint URL. In case of k8s controllers and operators, the URL points to the cluster API.
+Also used to uniquely identifies an endpoint within the scope of a service`, func() {
+			Format(FormatURI)
+			Default("")
+			MaxLength(200)
+			Example("https://mlflow.10.120.130.140.nip.io")
+		})
+	tag++
+	Field(tag, "extension_id", String, "Reference to the extension this endpoint belongs to", func() {
+		Pattern(identifierPattern)
+		Default("")
+		MaxLength(100)
+		Example("s3-storage-axW45s")
+	})
+	tag++
+	Field(tag, "service_id", String, "Reference to the service this endpoint belongs to", func() {
+		Pattern(identifierPattern)
+		Default("")
+		MaxLength(100)
+		Example("s3")
+	})
+	tag++
+	Field(tag, "endpoint_type", String,
```

Comment:
This should be just `"type"`, right?

Reply:
`type` is a reserved word in golang. I would rather avoid triggering it and just use this name instead. But yes, should be just `type`.

Reply:
Well, you are using `type` in the yaml example for credentials...

Reply:
Ok, done